include/vast/Interfaces/ClangAST/ASTDeclInterface.td

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2024-present, Trail of Bits, Inc.
+
+#ifndef VAST_INTERFACES_CLANG_AST_DECL_INTERFACE
+#define VAST_INTERFACES_CLANG_AST_DECL_INTERFACE
+
+include "mlir/IR/OpBase.td"
+include "vast/Interfaces/Common.td"
+
+def ASTTagDeclInterface : VastOpInterface< "ASTTagDeclInterface" > {
+    let description = [{
+        Inteface for clang's TagDecl AST node.
+    }];
+
+    let methods = [
+
+    ];
+}
+
+def ASTDeclInterface : VastOpInterface< "ASTDeclInterface" > {
+    let description = [{
+        Interface for clang's Decl AST node.
+    }];
+
+    let methods = [
+        InterfaceMethod< [{ Inditaces whether the declaration was implicitly
+                            generated by the implementation. If false, this
+                            declaration was written explicitly in the source code. }],
+                        "bool", "isImplicit", (ins) >,
+
+        InterfaceMethod< "", "void", "getDeclContext", (ins) >
+    ];
+}
+
+def ASTNamedDeclInterface : VastOpInterface< "ASTNamedDeclInterface",
+    [ASTDeclInterface] > {
+
+    let description = [{
+        Interface for clang's NamedDecl AST node.
+    }];
+
+    let methods = [];
+}
+
+def ASTValueDeclInterface : VastOpInterface< "ASTValueDeclInterface",
+    [ASTNamedDeclInterface] > {
+
+    let description = [{
+        Interface for clang's ValueDecl AST node.
+    }];
+
+    let methods = [
+        InterfaceMethod< "", "void", "getType", (ins) >
+    ];
+}
+
+def ASTDeclaratorDeclInterface : VastOpInterface< "ASTDeclaratorDeclInterface",
+    [ASTValueDeclInterface] > {
+    
+    let description = [{
+        Interface for clang's DeclaratorDecl AST node.
+    }];
+
+    let methods = [];
+}
+
+def ASTFieldDeclInterface : VastOpInterface< "ASTFieldDeclInterface",
+    [ASTDeclaratorDeclInterface] > {
+    let description = [{
+        Interface for clang's FieldDecl AST node.
+    }];
+
+    let methods = [
+        InterfaceMethod< "Determines whether this is an unnamed bitfield",
+        "bool", "isUnnamedBitField", (ins) >,
+
+        InterfaceMethod< [{ Determine if this field is a subobject of zero size,
```

Comment:
No need for long comments just always mention, that it mirror the AST method, e.g., `clang::FieldDecl::isZeroSize`. We don't have capacity to maintain comments them afterwards.

Reply:
Update this elsewhere as well.